devito/types/dense.py

```diff
@@ -408,6 +408,26 @@ def data(self):
         """
         return self.data_domain
 
+    def data_gather(self, start=None, stop=None, step=1, rank=0):
+        """
+        Gather distributed `Data` attached to a `Function` onto a single rank.
+
+        Parameters
+        ----------
+        rank : int
+            The rank onto which the data will be gathered.
+        step : int or tuple of ints
+            The `slice` step in each dimension.
+        start : int or tuple of ints
+            The `slice` start in each dimension.
+        stop : int or tuple of ints
+            The final point of the `slice` to include.
+        Notes
+        -----
+        Notes will go here.
```

Comment:
TODO

maybe also add something along the lines of "users should be aware that calling this method might blow up the memory since... so use it judiciously" -- what do you say?